wpimath/src/main/java/edu/wpi/first/math/Pair.java

```diff
@@ -60,4 +62,23 @@ public static <A, B> Pair<A, B> of(A a, B b) {
   public String toString() {
     return String.format("Pair(%s, %s)", m_first, m_second);
   }
+
+  /**
+   * Checks equality between this Pair and another object.
+   *
+   * @param obj The other object.
+   * @return Whether the two objects are equal or not.
+   */
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof Pair) {
+      return m_first.equals(((Pair) obj).getFirst()) && m_second.equals(((Pair) obj).getSecond());
```

Comment:
As currently written, this will NPE if m_first or m_second is null.  Objects.equals() can be used on each element to avoid this.